src/edgepi/adc/edgepi_adc.py

```diff
@@ -131,52 +128,74 @@ def __get_current_rtd_state(self) -> dict[str, PropertyValue]:
         return {
             "adc_1_analog_in": self.adc_1.mux_p.code,
             "adc_1_mux_n": self.adc_1.mux_n.code,
+            "adc_2_analog_in": self.adc_2.mux_p.code,
+            "adc_2_mux_n": self.adc_2.mux_n.code,
             "idac_1_mux": self.__get_state(ADCProperties.IDAC1_MUX).code,
             "idac_2_mux": self.__get_state(ADCProperties.IDAC2_MUX).code,
             "idac_1_mag": self.__get_state(ADCProperties.IDAC1_MAG).code,
             "idac_2_mag": self.__get_state(ADCProperties.IDAC2_MAG).code,
             "pos_ref_inp": self.__get_state(ADCProperties.REFMUX_POS).code,
             "neg_ref_inp": self.__get_state(ADCProperties.REFMUX_NEG).code,
+            "adc2_ref_inp": self.__get_state(ADCProperties.ADC2_REFMUX).code,
         }
 
-    def __get_rtd_state(self) -> bool:
+    def __get_rtd_state(self):
         """
         Get on/off RTD state.
+        Returns:
+            RTD
         """
-        return self.__get_current_rtd_state() == RTDModes.RTD_ON.value
+        # This is assuming, idac_1_mux - adc2_ref_inp are modified only when RTD is enabled.
+        # compare the current rtd_state to RTDMode.RTD_OFF.value dictionary. If it doesn't match,
+        # it means RTD is enabled
+        rtd_state = self.__get_current_rtd_state()
+        is_rtd_off = all(rtd_state.get(key)== value for key,value in RTDModes.RTD_OFF.value.items())
+        if is_rtd_off:
+            return RTDModes.RTD_OFF.value
+
+        if all(rtd_state.get(key) == value for key, value in\
+               (RTDModes.RTD_ON.value | ADC1RtdConfig.ON.value).items()):
+            return RTDModes.RTD_ON.value | ADC1RtdConfig.ON.value
 
+        if all(rtd_state.get(key) == value for key, value in\
+              (RTDModes.RTD_ON.value | ADC2RtdConfig.ON.value).items()):
+            return RTDModes.RTD_ON.value | ADC2RtdConfig.ON.value
```

Comment:
1. To keep the return type the same i'm returning value, but this can be resolved if we go with the alternate solution suggested below.
2. We decided to return the mode instead of boolean. I'm taking unions to pass which mode the RTD is on, alternatively, I can just return RTDModes.RTD_ON and do the union stuff as needed in the module.